tests/conftest.py

```diff
@@ -72,12 +76,77 @@ def firefox_notifications(notifications):
     params=[DESKTOP],
     ids=['Desktop'],
 )
-def selenium(selenium, request):
-    """Fixture to set a custom resolution for tests running on Desktop."""
+def selenium(selenium, base_url, session_auth, request):
+    """Fixture to set a custom resolution for tests running on Desktop
+    and handle browser sessions when needed"""
     selenium.set_window_size(*request.param)
+    # establishing actions  based on markers
+    create_session = request.node.get_closest_marker('create_session')
+    login = request.node.get_closest_marker('login')
+    clear_session = request.node.get_closest_marker('clear_session')
+    # this is used when we want to open an AMO page with a sessionid
+    # cookie (i.e. a logged-in user) already set
+    if create_session:
+        # need to set the url context if we want to apply a cookie
+        # in order to avoid InvalidCookieDomainException error
+        selenium.get(base_url)
+        # set the sessionid cookie
+        selenium.add_cookie(
+            {
+                "name": "sessionid",
+                "value": session_auth,
+            }
+        )
+    # this is used when we want to start the browser with a normal login
+    # mostly used for the scope of getting the session cookie and storing it for later use
+    if login:
+        home = Home(selenium, base_url).open().wait_for_page_to_load()
+        home.header.click_login()
+        home.wait.until(
+            EC.visibility_of_element_located((By.NAME, 'email')),
+            message=f'FxA email input field was not displayed in {selenium.current_url}',
+        )
+        user = login.args[0]
+        Login(selenium, base_url).account(user)
+        home.wait.until(
+            EC.url_contains('addons'),
+            message=f'AMO could not be loaded in {selenium.current_url}',
+        )
+        session_cookie = selenium.get_cookie('sessionid')
+        with open(user + '.txt', 'w') as file:
+            file.write(session_cookie['value'])
+    # delete the user session and files created for a test suite;
+    # this is normally used as the last step in a test suite
+    if clear_session:
+        # clear session
+        home = Home(selenium, base_url).open().wait_for_page_to_load()
+        home.header.click_logout()
+        # delete the file
+        user_file = create_session.args[0]
+        import os
+
+        if os.path.exists(f'{user_file}.txt'):
+            os.remove(f'{user_file}.txt')
+        else:
+            print("The file does not exist")
```

Comment:
My idea was to invoke this fixture marker only at the end of the test suit, to handle the logout (clearing session) and file deletion part. There wouldn't be other code to execute, except for verifying that the file is indeed removed. Below is an example.: I've tested  and debugged it extensively and it does what I'm expecting it to do - i.e. sets the session cookie, logs the user out and finally deletes the file.

```
@pytest.mark.create_session('regular_user')
@pytest.mark.clear_session
def test_clear_session(selenium, base_url):
    with pytest.raises(FileNotFoundError):
        with open('regular_user.txt', 'r') as file:
            file.read()
```

For the rest of the time, these fixture markers won't be used together in the same test, so there is no priority handling needed.

Reply:
If you meant the above test, it runs at the end of the suite because I'm placing it there (keep in mind that the submission tests will have to run in serial order).
If you're asking about the `@pytest.mark.clear_session` marker from the `selenium` fixture, this runs last, but it only runs if I'm specifically adding the marker to a test, otherwise it's ignored - and we want it to be ignored until the end of the test session.